modules/dkimVerifier.jsm

```diff
@@ -1209,7 +1233,23 @@ var Verifier = (function() {
 
 		// verify Signature
 		var keyInfo = {};
-		var isValid = verifyRSASig(DKIMSignature.DKIMKey.p, headerHashInput,
+		
+		// Selecting verification function depending on used signature algorithm
+		var verifyFunction = null;
+		switch (DKIMSignature.a_sig) {
+			case "ed25519":
+				verifyFunction = verifyED25519Sig;
+				headerHashInput = dkim_hash(headerHashInput, DKIMSignature.a_hash, "b64");
```

Comment:
Fine with leaving as is, but I think it would be more consistent to the RSA part to move this line into the `verifyED25519Sig()` function.

Reply:
Well, for now, the hash algorithm is always sha256, so I could hard code it in verifyED25519Sig function. But if this changes in future, I'll have to change all the verify functions' signatures to pass the hash algorithm. So I decided to place it here to have minimum  changes.
But I can also put it in the verify function, if you like it better. I had it first in the verify function, too...

Reply:
No you can leave it as is. Also forgot that the `verifyRSASig()` does not take the hash algorithm.